fdm-calculator/src/balance/nitrogen/input.ts

```diff
@@ -0,0 +1,139 @@
+                b_id_farm,
+                timeframe,
+            )
+
+            // Collect the details per field
+            const fields = await Promise.all(
+                farmFields.map(async (field) => {
+                    // Collect the cultivations of the field
+                    const cultivations = await getCultivations(
+                        tx,
+                        principal_id,
+                        field.b_id,
+                        timeframe,
+                    )
+
+                    // Collect the harvests of the cultivations
+                    // Collect a promise per cultivation
+                    const harvestPromises = cultivations.map(async (cultivation) => {
+                        return await getHarvests(
+                            tx,
+                            principal_id,
+                            cultivation.b_lu,
+                            timeframe,
+                        )
+                    })
+
+                    // Wait for all, then flatten the resulting arrays into one list
+                    const harvestArrays = await Promise.all(harvestPromises)
+                    const harvests = harvestArrays.flat()
+                    const harvestsFiltered = harvests.filter(
+                        (harvest) => harvest.b_lu !== undefined,
+                    )
+
+                    // Get the soil analyses of the field
+                    const soilAnalyses = await getSoilAnalyses(
+                        tx,
+                        principal_id,
+                        field.b_id,
+                        timeframe,
+                    )
+
+                    // Get the fertilizer applications of the field
+                    const fertilizerApplications =
+                        await getFertilizerApplications(
+                            tx,
+                            principal_id,
+                            field.b_id,
+                            timeframe,
+                        )
+
+                    return {
+                        field: field,
+                        cultivations: cultivations,
+                        harvests: harvestsFiltered,
+                        fertilizerApplications: fertilizerApplications,
+                        soilAnalyses: soilAnalyses,
+                    }
+                }),
+            )
+
+            // Collect the details of the fertilizers
+            const fertilizerDetails = await getFertilizers(
+                tx,
+                principal_id,
+                b_id_farm,
+            )
+
+            // Collect the details of the cultivations
+            const cultivationDetails = await getCultivationsFromCatalogue(
+                tx,
+                principal_id,
+                b_id_farm,
+            )
+
+            return {
+                fields,
+                fertilizerDetails: fertilizerDetails,
+                cultivationDetails,
+                timeFrame: timeframe,
```

Comment:
is there a reason why the return for cultivationDetails differ from fertilizerDetails?

Reply:
No, but for consistency I adjusted it to `cultivationDetails: cultivationDetails` at c0478d76d35329d3ef7647d4f60d2f1e6fb7ec3d